### ISSUE TITLE
Different store to cache strategies

### ISSUE THREAD

**Issue by vicmosin** [Kevinrob/guzzle-cache-middleware#51]
I noticed that every call the cache [being updated](https://github.com/Kevinrob/guzzle-cache-middleware/blob/master/src/CacheMiddleware.php#L212). Maybe it make sense to introduce different strategies of storing cache? For example, we use Etag as a cache mechanism so it's really useless to update cache everytime server responses with 304

**Comment by Kevinrob** [comment]
Hi, I see what you mean. But this will break the RFC.
```
[...] the cache MUST:
[...]
   o  use other header fields provided in the 304 (Not Modified)
      response to replace all instances of the corresponding header
      fields in the stored response.
```
http://tools.ietf.org/html/rfc7234#section-4.3.4

I understand, that a server can (by the way of a `304`) replace one or more headers in the cached response. So, it's not "really useless" in general.

Maybe, we can do a different method in `CacheStrategyInterface` like `update(RequestInterface $request, ResponseInterface $response)`. With that, you can extend the strategy and change the behavior of this method.

Is it a performance issue for you?

**Comment by vicmosin** [comment]
Yes, additional method will help as well.

**Comment by vicmosin** [comment]
In my case `Etag = md5(content)` so it is really meaningless to update the cache per each request.. Frontend is a WP site, so yes, it's a performance issue for me ))

**Comment by Kevinrob** [comment]
Ok, I will try to look at this today or tomorrow. Thank for reporting this issue!

**Comment by Kevinrob** [comment]
@vicmosin Can you check #52? Is it useful for you like that?

**Comment by Kevinrob** [comment]
Fixed with [v1.3](https://github.com/Kevinrob/guzzle-cache-middleware/releases/tag/v1.3).
